pptx_blueprint/__init__.py

```diff
@@ -54,7 +54,27 @@ def _find_shapes(self, label: str) -> Iterable[BaseShape]:
         Args:
             label (str): label of the placeholder (without curly braces)
         """
-        pass
+        slide_number, tag_name = label.split(":")
+        matched_shapes = []
+
+        def _find_shapes_in_slide(slide):
+            for shape in slide.shapes:
+                if shape.text == f'{{{tag_name}}}':
+                    matched_shapes.append(shape)
+
+        if slide_number == '*':
+            for slide in self._presentation.slides:
+                _find_shapes_in_slide(slide)
+        else:
+            # in label we are using 1 based indexing
+            slide_index = int(slide_number)-1
+            if slide_index < 0 or slide_index >= len(self._presentation.slides):
+                raise IndexError(f"Can't find slide number {slide_number}.")
+
+            slide = self._presentation.slides[slide_index]
+            _find_shapes_in_slide(slide)
```

Comment:
This duplication strikes me as odd. My approach would have been to wrap a single slide_index into an iterable. This way, you can handle multiple slides and one single slide on an equal footing.